pkg/apis/core/v1beta1/types.go

```diff
@@ -362,6 +383,124 @@ type StorageProfileList struct {
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type DataSourceList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	// Items provides a list of DataSources
+	Items []DataSource `json:"items"`
+}
+
+// DataImportCron defines a cron job for recurring polling/importing disk images as PVCs into a golden image namespace
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:storageversion
+type DataImportCron struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   DataImportCronSpec   `json:"spec"`
+	Status DataImportCronStatus `json:"status,omitempty"`
+}
+
+// DataImportCronSpec defines specification for DataImportCron
+type DataImportCronSpec struct {
+	// Source specifies where to poll disk images from
+	Source DataImportCronSource `json:"source"`
+	// Schedule specifies in cron format when and how often to look for new imports
+	Schedule string `json:"schedule"`
+	// GarbageCollectOutdated specifies whether old PVCs should be cleaned up after a new PVC is imported. Defaults to false.
+	// +optional
+	GarbageCollectOutdated *bool `json:"garbageCollectOutdated,omitempty"`
+	// ManagedDataSource specifies the name of the corresponding DataSource this cron will manage
+	ManagedDataSource string `json:"managedDataSource"`
+}
+
+// DataImportCronSource defines where to poll and import disk images from
+type DataImportCronSource struct {
+	Registry *DataImportCronSourceRegistry `json:"registry"`
+}
+
+// DataImportCronSourceRegistry provides the parameters to refer a registry source
+type DataImportCronSourceRegistry = DataVolumeSourceRegistry
+
+// DataImportCronStatus provides the most recently observed status of the DataImportCron
+type DataImportCronStatus struct {
+	// LastImportedPVC is the name of the last imported PVC
+	LastImportedPVC *string `json:"lastImportedPVC,omitempty"`
+	// LastExecutionTimestamp is the time of the last polling
+	LastExecutionTimestamp *string `json:"lastExecutionTimestamp,omitempty"`
+	// LastImport is the time of the last import
+	LastImport *string                   `json:"lastImport,omitempty"`
+	Conditions []DataImportCronCondition `json:"conditions,omitempty" optional:"true"`
+}
+
+// DataImportCronCondition represents the state of a data import cron condition
+// FIXME: needs LastTransitionTime/LastHeartbeatTime/Reason?
+type DataImportCronCondition struct {
+	Type    DataImportCronConditionType `json:"type" description:"type of condition ie. Progressing, UpToDate"`
+	Status  corev1.ConditionStatus      `json:"status" description:"status of the condition, one of True, False, Unknown"`
+	Message string                      `json:"message,omitempty" description:"human-readable message indicating details about last transition"`
+}
```

Comment:
fyi, there was some further discussion about DataImportCron conditions that can be found [here](https://github.com/kubevirt/community/pull/99#discussion_r639171296)

I think the big takeaway is that we need to be able to express when the DataImportCron is currently executing (Progressing = true|false) and whether it is up-to-date or not (UpToDate = true|false). It's okay if the condition naming changes in the final implementation, we just need a way to represent those two states somehow and have messages that indicate why the condition is set to true|false.

Reply:
Note all is mentioned in the field description.